qiskit/visualization/circuit_visualization.py

```diff
@@ -335,6 +343,11 @@ def _text_circuit_drawer(
     text_drawing.vertical_compression = vertical_compression
 
     if filename:
+        if not filename.endswith(".txt"):
+            raise ValueError(
```

Comment:
Qiskit raises `QiskitError` exceptions, I think in this case `VisualizationError` would be okey. 
```suggestion
            raise VisualizationError(
```